notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb

```diff
@@ -0,0 +1,416 @@
+    "output_keys = list(exec_net_onnx.outputs.keys())\n",
+    "\n",
+    "# get network input size\n",
+    "input_size = exec_net_onnx.input_info[input_keys[0]].input_data.shape[1]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "48eb25e2",
+   "metadata": {},
+   "source": [
+    "Defining a softmax function to extract the prediction from the output of the IR format."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "de01fccc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def softmax(x):\n",
+    "    e_x = np.exp(x - np.max(x))\n",
+    "    return e_x / e_x.sum() \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e778507",
+   "metadata": {},
+   "source": [
+    "# Inference"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cd429d55",
+   "metadata": {},
+   "source": [
+    "Creating a generic inference function to read the input and infer the result into 2 classes: Positive or Negative."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "cc0c91a6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def infer(input_text):\n",
+    "    input_text = tokenizer(input_text, padding='max_length', max_length = 512, truncation=True, return_tensors=\"pt\")\n",
+    "    inputs = {'input_ids': input_text['input_ids'], 'attention_mask': input_text['attention_mask']}\n",
+    "    result = exec_net_onnx.infer(inputs= inputs)\n",
+    "    result_ir = result[output_layer_onnx]\n",
+    "    probability = np.argmax(softmax(result_ir))\n",
+    "    print(\"Label:\")\n",
+    "    label = {0: 'NEGATIVE', 1: 'POSITIVE'}     \n",
+    "    return label[probability]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b60e79fd",
+   "metadata": {},
+   "source": [
+    "For a single input sentence"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "cf976f71",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "User Input:\n",
+      "Have a great day\n",
+      "Label:\n",
+      "POSITIVE\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"User Input:\")\n",
+    "input_text = input()\n",
+    "result = infer(input_text)\n",
+    "print(result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "29b4d013",
+   "metadata": {},
+   "source": [
+    "Read from a file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "id": "63f57d28",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "User Input:\n",
+      " The food was horrible.\n",
+      "Label:\n",
+      "NEGATIVE\n"
+     ]
+    }
+   ],
+   "source": [
+    "with open(\"data/sample.txt\",\"r\") as f:\n",
+    "    input_text = f.readlines()\n",
+    "print(\"User Input:\\n\",input_text[0])\n",
+    "result = infer(input_text)\n",
+    "print(result)"
```

Comment:
probably, if user will have several lines in file, it will break this sample, maybe it make sense to run in cycle line-by-line and print result for each line?